packages/local_auth/android/src/main/java/io/flutter/plugins/localauth/LocalAuthPlugin.java

```diff
@@ -72,6 +74,16 @@ public void onError(String code, String error) {
                 }
               });
       authenticationHelper.authenticate();
+    } else if (call.method.equals("getAvailableBiometrics")) {
+      FingerprintManager fingerprintMgr =
+          registrar.activity().getSystemService(FingerprintManager.class);
+      ArrayList<String> biometrics = new ArrayList<String>();
+      if (!fingerprintMgr.hasEnrolledFingerprints()) {
+        biometrics.add("undefined");
+      } else if (fingerprintMgr.isHardwareDetected() && fingerprintMgr.hasEnrolledFingerprints()) {
```

Comment:
`isHardwareDetected` is a device capability call. You should probably check that first:

```java
if (isHardwareDetected) {
  if (hasEnrolledFingerprint) {
    biometrics.add("fingerprint");
  } else {
    biometrics.add("undefined");
  }
}
result.success(biometrics);
```